modules/miner/cpuminer.go

```diff
@@ -20,8 +27,15 @@ func (m *Miner) threadedMine() {
 	// occurring.
 	cycleStart := time.Now()
 	for {
-		// Kill the thread if mining has been turned off.
 		m.mu.Lock()
+		// Kill the thread if the miner's threadgroup is stopped.
+		if m.tg.IsStopped() {
+			m.miningOn = false
+			m.mining = false
```

Comment:
behavior of exported methods after shutdown is undefined (at least when they can't return an error), so for correctness' sake, `CPUMining` should return false after shutdown.